lib-api/src/main/java/org/ergoplatform/appkit/BoxOperations.java

```diff
@@ -1,119 +1,159 @@
 package org.ergoplatform.appkit;
 
+import static com.google.common.base.Preconditions.checkState;
+import static org.ergoplatform.appkit.BlockchainContext.DEFAULT_LIMIT_FOR_API;
+import static org.ergoplatform.appkit.Parameters.MinFee;
+
+import com.google.common.base.Preconditions;
+
 import org.ergoplatform.P2PKAddress;
 import org.ergoplatform.appkit.impl.ErgoTreeContract;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+import java.util.function.Function;
 
-import static com.google.common.base.Preconditions.checkState;
-import static org.ergoplatform.appkit.Parameters.MinFee;
+import javax.annotation.Nonnull;
 
 /**
- * A collection of utility operations implemented in terms of abstract Appkit interfaces.
+ * A collection of utility operations for selecting boxes and building transactions
```

Comment:
This was the key point actually. By restricting the implementation to be on top of interfaces only, decouples this methods' logic from the implementation of the core Appkit interfaces.
A property worth preserving.

Reply:
The idea is to clearly decouple Impl classes (or hide them) from the code, which uses abstract interfaces. Basically any code that uses interfaces alone is abstracted from the concrete detail of how those interfaces are implemented in Impl classes. 
Current implementation of BoxOperations is good enough, as it doesn't use Impl classes and is written on top of the interfaces.
But I suggest to leave the phrase `implemented in terms of abstract Appkit interfaces.` in place.

Reply:
Done.